gulpfile.js

```diff
@@ -83,6 +84,41 @@ const common = {
       path.join(outputPath, 'js/**'),
       path.join(outputPath, 'css/**')
     ], { ignoreInitial: true }, common.syncJS_CSSinRt)
+  },
+
+  async env (cb) {
+    const mode = process.env.PLATFORM || 'ios'
+
+    const IP = ip.address()
+    const LOGSEQ_APP_SERVER_URL = `http://${IP}:3001`
+
+    if (typeof global.fetch === 'function') {
+      try {
+        await fetch(LOGSEQ_APP_SERVER_URL)
+      } catch (e) {
+        return cb(new Error(`/* ❌ Please check if the service is ON. (${LOGSEQ_APP_SERVER_URL}) ❌ */`))
+      }
+    }
+
+    console.log(`------ Cap ${mode.toUpperCase()} -----`)
+    console.log(`Dev serve at: ${LOGSEQ_APP_SERVER_URL}`)
+    console.log(`--------------------------------------`)
+
+    cp.execSync(`npx cap sync ${mode}`, {
+      stdio: 'inherit',
+      env: Object.assign(process.env, {
+        LOGSEQ_APP_SERVER_URL
+      })
+    })
+
+    cp.execSync(`npx cap run ${mode} --external`, {
```

Comment:
This is more of a personal development habit. 
I always use IDE.😅 So this command annoys me because I need to kill it halfway.

Reply:
> This is more of a personal development habit. I always use IDE.😅 So this command annoys me because I need to kill it halfway.

Yes, it is difficult to cover most development workflows with one command. In fact, the key work is that we do not have to customize the configuration by modifying [capacitor.config.ts](https://github.com/logseq/logseq/pull/6545/files#diff-cc7bbd9d83faa38b4df37d38bfa2a22f9154663864dec880283b29423be70cc9) every time, while being prone to accidentally commit this file to the remote.

Of course, if this default command doesn't satisfy the needs, we can customize it ourselves, as what's the workflow of mine.

```shell
# prepare dev envs
export MY_IP=$(LANG=C /sbin/ifconfig  | sed -ne $'/127.0.0.1/ ! { s/^[ \t ]*inet[ \t ]\\{1,99\\}\\(addr:           \\)\\{0,1\\}\\([0-9.]*\\)[ \t\/ ].*$/\\2/p;  }') 
export LOGSEQ_APP_SERVER_URL="http://${MY_IP}:3001"

# commands
alias logseq-ios-dev='npx cap sync && npx cap ios'
alias logseq-android-dev='npx cap sync && npx cap android'
````